src/painter.rs

```diff
@@ -130,109 +128,145 @@ impl Painter {
             );
         }
 
-        // TODO: make a PR with repr(c) on Vertex and just use "mesh.vertices"
-        let vertices = mesh
-            .vertices
-            .iter()
-            .map(|x| Vertex {
-                pos: (x.pos.x, x.pos.y),
-                uv: x.uv,
-                color: (x.color.0[0], x.color.0[1], x.color.0[2], x.color.0[3]),
-            })
-            .collect::<Vec<Vertex>>();
-        self.bindings.vertex_buffers[0].update(ctx, &vertices);
+        self.bindings.vertex_buffers[0].update(ctx, &mesh.vertices);
+
+        let screen_size = ctx.screen_size();
 
         // TODO: support u32 indices in miniquad and just use "mesh.indices"
```

Comment:
If you did not looked into this yet - I can do it

Reply:
I added a call to the [`split_to_u16()`](https://docs.rs/epaint/0.8.0/epaint/struct.Triangles.html#method.split_to_u16) so 32-bit indices is not a very high priority. The only thing that could break if is somebody actually adds a triangle mesh to the egui painter that uses more than 2^16 vertices, but normal egui usage will be fine.